python/ray/rllib/utils/tf_policy_graph.py

```diff
@@ -34,25 +35,31 @@ class TFPolicyGraph(PolicyGraph):
     """
 
     def __init__(
-            self, observation_space, action_space, sess, obs_input,
-            action_sampler, loss, loss_inputs,
-            is_training, state_inputs=None, state_outputs=None):
+            self, sess, observation_space, action_space, obs_input,
+            action_sampler, loss, loss_inputs, is_training,
+            state_inputs=None, state_outputs=None, seq_lens=None,
+            max_seq_len=20):
         """Initialize the policy graph.
 
         Arguments:
+            sess (Session): TensorFlow session to use.
             observation_space (gym.Space): Observation space of the env.
             action_space (gym.Space): Action space of the env.
-            sess (Session): TensorFlow session to use.
-            obs_input (Tensor): input placeholder for observations.
-            action_sampler (Tensor): Tensor for sampling an action.
+            obs_input (Tensor): input placeholder for observations, of shape
+                [BATCH_SIZE, obs...].
+            action_sampler (Tensor): Tensor for sampling an action, of shape
+                [BATCH_SIZE, action...]
             loss (Tensor): scalar policy loss output tensor.
             loss_inputs (list): a (name, placeholder) tuple for each loss
                 input argument. Each placeholder name must correspond to a
-                SampleBatch column key returned by postprocess_trajectory().
+                SampleBatch column key returned by postprocess_trajectory(),
+                and has shape [BATCH_SIZE, data...].
             is_training (Tensor): input placeholder for whether we are
                 currently training the policy.
             state_inputs (list): list of RNN state output Tensors.
             state_outputs (list): list of initial state values.
+            seq_lens (Tensor): placeholder for RNN sequence lengths.
```

Comment:
can you add shape for this tensor? they are very helpful

Reply:
Done